registry/api/v2/urls.go

```diff
@@ -157,6 +157,18 @@ func (ub *URLBuilder) BuildBlobURL(ref reference.Canonical) (string, error) {
 	return layerURL.String(), nil
 }
 
+// BuildBlobRelativeURL constructs the relative url for the blob identified by name and dgst.
+func (ub *URLBuilder) BuildBlobRelativeURL(ref reference.Canonical) (string, error) {
```

Comment:
`BuildBlobURL` returns a string, so making relative would involve re-parsing into a URL and stripping off the non-relative parts.

The problem with URL builder is that cloned URLs always have  `http.ResolveReference`  called on them.  Though I think this second approach is preferable.